packages/opentelemetry-core/src/trace/TraceState.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+
+// TODO validate maximum number of items
+// const maximumTraceStateItems = 32;
+
+const maximumTraceStateLength = 512;
+export interface InternalTraceState {
+  [key: string]: string;
```

Comment:
Why is this limited to strings? We store other types in the trace state.

Reply:
I think this is easier for serialization/deserialization and this is how other languages are keeping it. What are the types do you use/prefer in the trace state?

Reply:
The problem is if other types of object are stored and manipulated during the lifetime of the local trace, then conversion will constantly be needed. This is generally a concern for anywhere that accepts only string values, but in this case it may be fine since the trace state properties are usually not altered after they are set.

Reply:
Actually, if this is only used at propagation time to another service, my above concern doesn't exist. This is especially true if there is a single trace state by vendor.